GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1367,19 +1368,19 @@ void OnRevisionReadCompleted()
                         await semaphoreUpdateGrid.WaitAsync(cancellationToken);
                     }
 
-                    IEnumerable<ObjectId> headParents = null;
+                    IReadOnlyList<ObjectId>? headParents = null;
                     bool refresh = false;
                     if (!headIsHandled && ShowArtificialRevisions())
                     {
                         if (CurrentCheckout is not null)
                         {
                             // Not found, so search for its parents
-                            headParents = TryGetParents(Module, _filterInfo, CurrentCheckout);
+                            headParents = TryGetParents(Module, _filterInfo, CurrentCheckout).ToList();
                         }
 
                         // If parents are rewritten HEAD may not be included
                         // Insert the artificial commits where relevant if possible, otherwise first
-                        refresh = AddArtificialRevisions(headParents);
+                        refresh = AddArtificialRevisions(headParents ?? Enumerable.Empty<ObjectId>().ToList());
```

Comment:
```suggestion
                        refresh = AddArtificialRevisions(headParents ?? new List<ObjectId>());
```
or create and use a static such empty list instance.

Reply:
This is an exception for a very uncommon scenario, so I will use the shortest suggestion.